### PR TITLE
feat(atomic): network improvements

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1620

First set of performance improvements

1. Start loading i18n configured language sooner, before `initialize` is called
2. Only load configured language for i18n, don't load fallback language (e.g. if "fr", "en.json" was still loaded)
3. Remove lodash imports (should only be used as a dev dep), saves ~97kb
4. Use loading="lazy" on `img` tags

### CODE REVIEW

File: packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx [hunk 1]
```diff
@@ -166,19 +171,17 @@ export class AtomicSearchInterface {
 
   @Watch('language')
   public updateLanguage() {
-    if (!this.engineIsCreated(this.engine)) {
-      return;
+    if (this.engineIsCreated(this.engine)) {
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx:174]: Why the change in this function ? I personally like the early return pattern, helps readability.

File: packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx [hunk 2]
```diff
@@ -235,7 +238,7 @@ export class AtomicSearchInterface {
   /**
    * Initializes the connection with the headless search engine using options for `accessToken` (required), `organizationId` (required), `renewAccessToken`, and `platformUrl`.
    */
-  @Method() public initialize(options: InitializationOptions) {
+  @Method() public async initialize(options: InitializationOptions) {
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx:241]: Why the async keyword if we do not use await in the body of the function ? 

It was returning a promise anyway before, no ?

File: packages/atomic/src/components/result-lists/result-list-common.tsx [hunk 3]
```diff
@@ -167,7 +166,9 @@ export class ResultListCommon {
     const templates = (
       includeDefaultTemplate ? [this.makeDefaultTemplate()] : []
     ).concat(
-      customTemplates.filter(identity) as ResultTemplate<DocumentFragment>[]
+      customTemplates.filter(
```
Comment (olamothe) [inline packages/atomic/src/components/result-lists/result-list-common.tsx:169]: What is this change for ?

File: packages/atomic/src/components/result-template-components/atomic-result-image/atomic-result-image.tsx [hunk 4]
```diff
@@ -53,6 +53,12 @@ export class AtomicResultImage implements InitializableComponent {
       return;
     }
 
-    return <img alt={`${this.field} image`} src={filterProtocol(url)} />;
+    return (
+      <img
+        alt={`${this.field} image`}
+        src={filterProtocol(url)}
+        loading="lazy"
```
Comment (olamothe) [inline packages/atomic/src/components/result-template-components/atomic-result-image/atomic-result-image.tsx:60]: Great idea !

**Review by olamothe — APPROVED** [review]
GG !

Any idea of much loading time this shaves off ? Relatively speaking (%)

**Comment by ThibodeauJF** [comment]
> Any idea of much loading time this shaves off ? Relatively speaking (%)

Depends on the connection (removing lodash improves a lot on 3G) and when the user trigger initialize, then the i18n changes have a slight difference.

On a very good connection, these network improvements don't do much. More improvements to come...

File: packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx [hunk 5]
```diff
@@ -337,7 +340,7 @@ export class AtomicSearchInterface {
     return this.i18n.use(Backend).init({
       debug: this.logLevel === 'debug',
       lng: this.language,
-      fallbackLng: ['en'],
+      fallbackLng: false,
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx:343]: If some strings are missing from the language set by the user, are english strings going to be used?

Comment (ThibodeauJF) [reply packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx:343]: Good point! let me check...

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx (hunk 5 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx:343]: This indeed should be an fr/en improvement
